static/js/narrow.js

```diff
@@ -386,6 +392,67 @@ exports.update_selection = function (opts) {
     unread_ops.process_visible();
 };
 
+exports.to_compose_target = function () {
+    if (compose_state.get_message_type() === 'stream' && compose_state.stream_name() !== '') {
+        var stream = compose_state.stream_name();
+        var topic = compose_state.subject();
+        exports.by_stream_topic_name(stream, topic);
+        return true;
+    }
+    if (compose_state.get_message_type() === 'private') {
+        exports.by_recipient_name(compose_state.recipient());
+        return true;
+    }
+    return false;
+};
+
+exports.by_stream_topic_name = function (stream, topic) {
+    var stream_id = stream_data.get_stream_id(stream);
+
+    // If stream is incorrect, we dont do anything.
+    if (stream_id === undefined) {
+        return;
+    }
+
+    var topics = topic_data.get_recent_names(stream_id);
+
+    var opts = {
+        select_first_unread: true,
```

Comment:
select_first_unread is no longer a thing.

Reply:
Handled